docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,261 @@
+
+==== Required Inputs
+
+From this, we reach the full set of required inputs and where they can be
+procured or computed:
+
+- N, the number of signers in the group, available on the Keep contract
+  from the `getMembers` view method, but also tracked locally on the client
+  alongside the members of the keep.
+- L, the deposit lot size, available on the tBTC deposit contract from the
+  `lotSizeSatoshis` view method.
+- The previous outpoint describing the deposit UTXO, available on the tBTC
+  deposit contract from the `fundingInfo` view method, which returns it as
+  the third `utxoOutpoint` entry in its 3-part return tuple.
+- The keep's public key, available on the Keep contract from the
+  `getPublicKey` view method.
+- The deposit redemption script, straightforwardly computable from the keep's
+  public key in a way detailed in the section on <<output-script>>.
+- One Bitcoin address per signer, specified on the client in a way detailed
+  in the section on <<recovery-address>>.
+- The keep's signature on the sighash produced by the previous items,
+  computed via the threshold ECDSA protocol in a way detailed in the section on
+  <<signing>>.
+- F, the desired fee, which is a function of the going feerate on the Bitcoin
+  network and the number of bytes in the final transaction.
+
+The desired fee is the most problematic one to procure under the goal of not
+strictly requiring a Bitcoin node connection, as generally a Bitcoin node
+provides estimates of the feerates that will achieve transaction confirmation
+within a certain amount of time. A short-term solution is proposed in the
+section on <<bitcoin-fee>>.
+
+[[output-script]]
+===== Computing the Deposit Output Script
+
+[[recovery-address]]
+===== Specifying an Operator's Bitcoin Recovery Address
+
+To specify a given operator's Bitcoin Recovery Address, this proposal
+suggests adding a single key to the `Extensions.TBTC.BTCRefunds` config
+entry, `BeneficiaryAddress`. The value for this key should be one of:
+
+- A simple BTC address.
+- An xpub, ypub, or zpub that allows the client to derive new addresses for
+  each refund.
+
+The challenge with a *pub is that, to avoid reuse, it needs to do one of two
+things:
+
+- Be able to check, on-chain, whether a given address has been used; or,
+- Have a *pub that is only meant for use by the client, and have the client
+  track the latest address generated (as an index starting at 0).
+
+To avoid _strictly requiring_ a connection to a Bitcoin node, this RFC
+proposes tracking the latest address generated, and allowing an additional
+config key, `TargetDerivationStartIndex`, that sets a starting point for such
+derivations. The client should write the updated index to the `current/`
+directory that tracks active key shares, to an encrypted file named
+`current-btc-refund-derivation-index`. The operator should then seek to
+obtain from the staker an extended public key that is only used for this
+purpose.
```

Comment:
Yeah, I didn't get a chance to address that piece in @pdyraga's notes above: I'm fairly opposed to having a contract in the mix here. We have a clear trigger for when to start this with existing contracts, and contracts introduce cost, adaptability, and maintainability concerns that I suggest we avoid unless we have a security motivation to do otherwise. The contracts can't enforce anything in this scenario anyway, so using them strictly for data coordination seems unnecessary. Using them for operator-specific data tracking is even less desirable IMO---that info should remain private.

Lastly, we probably shouldn't put a *pub on-chain, as it's considered less secure than an address by a fair margin. If we don't have the *pub on-chain, having the derivation index on-chain seems problematic, e.g. if the staker decides to give the operator a different address.